fmriprep/workflows/bold/base.py

```diff
@@ -607,19 +624,19 @@ def init_func_preproc_wf(bold_file):
                 ('outputnode.itk_bold_to_t1', 'transforms')]),
             (bold_t1_trans_wf, boldmask_to_t1w, [
                 ('outputnode.bold_mask_t1', 'reference_image')]),
-            (bold_bold_trans_wf, boldmask_to_t1w, [
-                ('outputnode.bold_mask', 'input_image')]),
             (boldmask_to_t1w, outputnode, [
                 ('output_image', 'bold_mask_t1')]),
+            (final_boldref_wf, boldmask_to_t1w, [
+                ('outputnode.bold_mask', 'input_image')])
         ])
 
     if nonstd_spaces.intersection(('func', 'run', 'bold', 'boldref', 'sbref')):
         workflow.connect([
-            (bold_bold_trans_wf, outputnode, [
-                ('outputnode.bold', 'bold_native')]),
-            (bold_bold_trans_wf, func_derivatives_wf, [
-                ('outputnode.bold_ref', 'inputnode.bold_native_ref'),
+            (final_boldref_wf, func_derivatives_wf, [
+                ('outputnode.ref_image', 'inputnode.bold_native_ref'),
                 ('outputnode.bold_mask', 'inputnode.bold_mask_native')]),
+            (bold_bold_trans_wf if not multiecho else bold_t2s_wf, outputnode, [
```

Comment:
This seems to be a semantic change, not just a reorganization. Was this something we were doing wrong?

Reply:
Sorry, yes, I think it was wrong before, since it took the BOLD files from `bold_bold_trans_wf` for the native-space BOLD outputs.

Reply:
Looking through, I agree with this change. If I'm reading it right, a native BOLD space derivative for MEEPI will be the optimal combination, which seems appropriate.